packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -1116,4 +1121,20 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         RegistryInterface registry = RegistryInterface(finder.getImplementationAddress(OracleInterfaces.Registry));
         require(registry.isContractRegistered(msg.sender) || msg.sender == migratedAddress, "Caller not registered");
     }
+
+    // Checks if a request should be deleted. A request should be deleted if it has been rolled more than the maxRolls
+    function _shouldDeleteRequest(uint256 rollCount, bool isGovernance) private view returns (bool) {
+        return rollCount > maxRolls && !isGovernance;
+    }
+
+    // Returns the actual roll count of a request. This is the roll count plus the number of rounds that have passed
+    // since the last voting round.
+    function _getActualRollCount(PriceRequest storage priceRequest, uint32 currentRoundId)
+        private
+        view
+        returns (uint32)
+    {
+        if (currentRoundId <= priceRequest.lastVotingRound) return priceRequest.rollCount;
+        return priceRequest.rollCount + currentRoundId - priceRequest.lastVotingRound;
+    }
```

Comment:
there are no gas implications from pulling these out to private/internal functions, right?

Reply:
TL;DR: Overall, the new gas consumptions seem to have slightly lower average gas consumption than the old ones, but the maximum gas consumption is either the same or slightly higher in some cases.

| Function | min  | max | avg |
|--|--|--|--|
| NEW : processResolvablePriceRequests | 36221 |10033176 | 398914|
| OLD: processResolvablePriceRequests | 36221 |10000676 | 408182|
| NEW: requestPrice | 60640 |176259 | 160237|
| OLD: requestPrice | 60662 |176281 | 160922|
| NEW: updateTrackers | 55614 |850173 | 127541|
| OLD: updateTrackers | 55636 |850195 | 127484|

The new `processResolvablePriceRequests` function has a maximum gas consumption of 10033176, which is slightly higher than the old version's maximum of 10000676. However, the average gas consumption of 398914 is lower than the old version's average of 408182.
requestPrice:

The new `requestPrice` function has a maximum gas consumption of 176259, which is the same as the old version's maximum of 176281. However, the average gas consumption of 160237 is lower than the old version's average of 160922.


The new `updateTrackers` function has a maximum gas consumption of 850173, which is the same as the old version's maximum of 850195. The average gas consumption of 127541 is slightly lower than the old version's average of 127484.